pyrgg/graph_gen.py

```diff
@@ -307,18 +334,64 @@ def tsv_maker(
         sign,
         direct,
         self_loop,
-        multigraph)
-    edge_dic = dicts[0]
-    weight_dic = dicts[1]
-    edge_number = dicts[2]
-    for i in edge_dic.keys():
-        for j, value in enumerate(edge_dic[i]):
-            file.write(str(i) + "\t" + str(value) + "\t" +
-                       str(weight_dic[i][j]) + "\n")
-    file.close()
+        multigraph,
+    )
+    with open(file_name + ".tsv", "w") as buf:
+        _write_separated_file(buf, edge_dic, weight_dic, separator='\t')
     return edge_number
 
 
+def _write_separated_file(buf, edge_dic, weight_dic, separator, prefix=''):
+    """Write data to buffer separated with ``separator``."""
+    dummy_prefix = object()
+    prefix = prefix or dummy_prefix
+
+    for key, edge_val in edge_dic.items():
+        for j, value in enumerate(edge_val):
+            elements = [
+                prefix,
+                str(key),
+                str(value),
+                str(weight_dic[key][j]) + "\n"
+            ]
+            string = separator.join(x for x in elements if x != dummy_prefix)
+            buf.write(string)
+
+
+def json_to_yaml(filename):
```

Comment:
I think you duplicate this function, exactly the same function is in `functions.py`

Reply:
Thank you. I removed the duplication.